src/hooks/useForm.ts

```diff
@@ -12,7 +14,7 @@ export interface ChangeEvent<TData = any> {
     value: TData;
   };
 }
-export type SubmitPromise = Promise<any[]>;
+export type SubmitPromise<TData = any> = Promise<TData>;
```

Comment:
Since our submits are not consistent and not always return an array, leaving this for now and will type all the SubmitPromises properly in the next pr